allennlp/common/checks.py

```diff
@@ -39,10 +39,16 @@ def check_dimensions_match(dimension_1: int,
 
 
 def check_for_gpu(device_id: int):
-    if device_id is not None and device_id >= cuda.device_count():
-        raise ConfigurationError("Experiment specified a GPU but none is available;"
-                                 " if you want to run on CPU use the override"
-                                 " 'trainer.cuda_device=-1' in the json config file.")
+    if device_id is not None:
+        devices_available = cuda.device_count()
+        if not devices_available:
```

Comment:
two readability nits:

I would call the variable `num_available` or something else indicating it's a count 

and also I would be explicit that we're checking for zero rather than just falsiness:

```
if num_available == 0:
```

Reply:
Fixed, thanks!